### PR TITLE
HIVE-25594: Setup JDBC databases in tests via QT options

### PR DESCRIPTION
**Pull request by zabetak**
### What changes were proposed in this pull request?
1. Refactor `AbstractExternalDB` and subclasses: (i) turn setters to getters; (ii) inline methods; (iii) reduce visibility when possible; (iv) drop redundant comments; (v) drop unnecessary logging and try/catch blocks.
2. Use provided user/password/dbname when starting Postgres instead of default.
3. Use correct driver and docker image for MySQL database.
4. Support additional databases (MariaDB, Oracle, MSSQLServer) for tests.
5. Introduce `QTestDatabaseHandler` class for handling completely setup, init and cleanup of dockerized DBMS.
6. Drop `TestMiniLlapExtDBCliDriver`, with associated config, properties etc. 

### Why are the changes needed?
It is now possible to:
1. initialize a dockerized DBMS in any qtest without the need for specific CLI driver or configuration;
2. use multiple DBMS of different types in the same qtest;
3. use different initialization scripts per test/database;
4. use additional databases (MariaDB, Oracle, MSSQLServer) in tests.

Misc:
5. Improve readability and encapsulation of `AbstractExternalDB` and friends.
6. Small bugs in Postgres and MySQL.

### Does this PR introduce _any_ user-facing change?
No, it only impacts developers.

### How was this patch tested?
`mvn test -Dtest=TestMiniLlapLocalCliDriver -Dqfile_regex="qt_database.*" -Dtest.output.overwrite`

### CODE REVIEW

File: itests/util/src/main/java/org/apache/hadoop/hive/ql/externalDB/AbstractExternalDB.java [hunk 1]
```diff
@@ -71,27 +59,8 @@ public ProcessResults(String stdout, String stderr, int rc) {
...
-        return abstractExternalDB;
-    }
-
-    public AbstractExternalDB(String externalDBType) {
-        this.externalDBType = externalDBType;
-    }
-
-    protected String getDockerContainerName() {
-        return String.format("qtestExternalDB-%s", externalDBType);
+    private final String getDockerContainerName() {
```
[the remark this answers is not in the document]
Comment (kgyrtkirk) [reply itests/util/src/main/java/org/apache/hadoop/hive/ql/externalDB/AbstractExternalDB.java:62]: sure; can be covered in a followup as well

File: itests/util/src/main/java/org/apache/hadoop/hive/ql/qoption/QTestDatabaseHandler.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+  @Override
+  public void beforeTest(QTestUtil qt) throws Exception {
+    if (databaseToScript.isEmpty()) {
+      return;
+    }
+    for (Map.Entry<DatabaseType, String> dbEntry : databaseToScript.entrySet()) {
+      String scriptsDir = QTestUtil.getScriptsDir(qt.getConf());
+      Path dbScript = Paths.get(scriptsDir, dbEntry.getValue());
+      AbstractExternalDB db = dbEntry.getKey().create();
```
[the remark this answers is not in the document]
Comment (kgyrtkirk) [reply itests/util/src/main/java/org/apache/hadoop/hive/ql/qoption/QTestDatabaseHandler.java:113]: something like:
* use some real class insted of the enum
* retain an object for the lifetime of the backing docker container to have an object represent it in the code
* give the instructions to that object

but this is not that important...we can have it this way as well

**Review by kgyrtkirk — APPROVED** [review]

File: itests/util/src/main/java/org/apache/hadoop/hive/ql/externalDB/AbstractExternalDB.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zabetak) [reply itests/util/src/main/java/org/apache/hadoop/hive/ql/externalDB/AbstractExternalDB.java:62]: Logged https://issues.apache.org/jira/browse/HIVE-25667 for tracking this further.

File: itests/util/src/main/java/org/apache/hadoop/hive/ql/qoption/QTestDatabaseHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zabetak) [reply itests/util/src/main/java/org/apache/hadoop/hive/ql/qoption/QTestDatabaseHandler.java:113]: This suggestion would be definitely very useful and necessary if we want to support multiple databases of the same type running side by side or enabling database reuse among qfiles. Leaving this for https://issues.apache.org/jira/browse/HIVE-25668 where we are going to have more concrete requirements.

**Pull request closed without merging**